cmd/charm/charmcmd/cmd.go

```diff
@@ -138,24 +139,56 @@ func newCharmStoreClient(ctxt *cmd.Context, auth authInfo) (*csClient, error) {
 			BakeryClient: bakeryClient,
 			User:         auth.username,
 			Password:     auth.password,
-		}),
+		}).WithChannel(channel),
 		jar:  jar,
 		ctxt: ctxt,
 	}
 	return &csClient, nil
 }
 
+// addChannelFlag adds the -c (--channel) flags to the given flag set.
+func addChannelFlag(f *gnuflag.FlagSet, cv *chanValue) {
+	chans := []string{
+		string(params.StableChannel),
```

Comment:
I wonder if the params package should export a variable with all the channels in.
Perhaps it should export OrderedChannels and ValidChannels.